actor/builtin/miner/miner.go

```diff
@@ -719,7 +726,7 @@ func (ma *Actor) GetPower(ctx exec.VMContext) (*types.BytesAmount, uint8, error)
 
 // SubmitPoSt is used to submit a coalesced PoST to the chain to convince the chain
 // that you have been actually storing the files you claim to be.
-func (ma *Actor) SubmitPoSt(ctx exec.VMContext, poStProofs []types.PoStProof) (uint8, error) {
+func (ma *Actor) SubmitPoSt(ctx exec.VMContext, poStProofs []types.PoStProof, done types.IntSet) (uint8, error) {
```

Comment:
QUESTION: I observe that [the spec uses the `BitField` type](https://github.com/filecoin-project/specs/blob/master/actors.md#submitpost) (an RLE+ encoded bit set) where you've used an `IntSet` (a slice of u64). Is this a temporary divergence from the spec (perhaps because we don't yet have a `BitField` type with corresponding set operations)? If not, would you upstream this change to the spec?

Reply:
This is not temporary and I have considered upstreaming changes to the spec.  The issue is that the spec doesn't really distinguish between encodings of VM data and internal VM datastructures and the Bitfield is just the encoding of an IntSet.  I've been told that trying to put specifics of VM internals (like how the IntSet is implemented) into the spec is out of scope because that gets into implementation choices.  So there is a fine line I don't have a good feel for.  I don't think I am the right person to start introducing this subtle distinction into the spec or that it would even be a welcome addition.

Reply:
> I've been told that trying to put specifics of VM internals (like how the IntSet is implemented) into the spec is out of scope because that gets into implementation choices.

I guess all that really matters is that go-filecoin can send and receive `submitPoSt` messages whose bytes are conformant with the specification (i.e. go-filecoin knows how to send/receive RLE+ bit sets, in this context). If we choose to transform the decoded bit set into a slice of uint64s (in Go), then that's an implementation detail of go-filecoin that doesn't need upstreaming. Does this overlap with what you're thinking, @ZenGround0?

Reply:
Yes this lines up.  And once the RLE+ PR lands all IntSet parameters will be RLE+ encoded.